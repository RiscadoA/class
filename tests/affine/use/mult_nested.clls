proc main() {
    cut {
        println("affine 1"); affine x;
        println("send 1"); send x(a. println("close 1"); close a);
        println("affine 2"); affine x;
        println("send 2"); send x(b. println("close 2"); close b);
        println("affine 3"); affine x;
        println("send 3"); send x(c. println("wait 3"); wait c; ());
        println("close 4"); close x
        |x: ~affine send close; affine send close; affine send wait; close|
        println("use 1"); use x;
        println("recv 1"); recv x(a);
        println("wait 1"); wait a;
        println("use 2"); use x;
        println("recv 2"); recv x(b);
        println("wait 2"); wait b;
        println("use 3"); use x;
        println("recv 3"); recv x(c);
        println("wait 4"); wait x;
        println("close 3"); close c
    }
};;
