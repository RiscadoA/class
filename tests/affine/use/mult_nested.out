use 1
affine 1
send 1
close 1
recv 1
wait 1
use 2
affine 2
send 2
close 2
recv 2
wait 2
use 3
affine 3
send 3
close 4
recv 3
wait 4
close 3
wait 3
